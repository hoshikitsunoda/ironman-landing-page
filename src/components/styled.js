import { createGlobalStyle } from 'styled-components'

// Global

export const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Nunito|Ropa+Sans&display=swap');

  html {
    background-color: #fffafa;
  }
  
  html * {
    font-family: 'Nunito', sans-serif;
  }

  h1,
  h2,
  h3,
  h4,
  p {
    margin: 0 auto;
  }

  h1,
  h2,
  h3,
  h4 {
    font-family: 'Ropa Sans', sans-serif;
  }

  img {
    display: block;
  }
`
