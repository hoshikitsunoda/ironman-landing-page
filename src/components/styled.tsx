import { createGlobalStyle } from 'styled-components'

// Global

export const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Nunito|Ropa+Sans&display=swap');

  html {
    background-color: #ffffff;
  }
  
  html * {
    font-family: 'Nunito', sans-serif;
  }

  body {
    margin: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  h1,
  h2,
  h3,
  h4,
  p {
    margin: 0 auto;
  }

  h1,
  h2,
  h3,
  h4 {
    font-family: 'Ropa Sans', sans-serif;
  }

  img {
    display: block;
  }
`

const size = {
  mobile: '350px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1200px',
}

export const Device = {
  mobile: `(min-width: ${size.mobile})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  laptopL: `(min-width: ${size.laptopL})`,
}
