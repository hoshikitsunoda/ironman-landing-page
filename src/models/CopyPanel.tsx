export class Panel {
  constructor(
    public icon: string,
    public heading: string,
    public body: string
  ) {}
}

interface PanelData {
  icon: string
  heading: string
  body: string
}

export interface PanelItem {
  [key: string]: PanelData
}
