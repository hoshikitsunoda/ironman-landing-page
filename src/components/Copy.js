import React from 'react'

import styled from 'styled-components'

const Copy = props => {
  const imgSrc = 'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55.jpg'
  return (
    <CopyWrapper>
      <HeadingWrapper className="text-wrapper heading">
        <p>Contrary to popular belief, he knows exactly what heâ€™s doing.</p>
      </HeadingWrapper>
      <div>
        <img src={imgSrc} alt="thumbnail" />
      </div>
    </CopyWrapper>
  )
}

const CopyWrapper = styled.div`
  padding: 0 1.2rem;

  img {
    width: 100%;
  }
`

const HeadingWrapper = styled.div`
  text-align: center;
  padding: 3rem 2.5rem;

  p {
    font-size: 1.5rem;
    line-height: 2.2rem;
  }
`

export default Copy
