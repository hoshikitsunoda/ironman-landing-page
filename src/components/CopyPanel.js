import React from 'react'

import styled from 'styled-components'

import GeniusIcon from '../images/icons/icon-genius.png'
import TacticianIcon from '../images/icons/icon-tactician.png'
import EngineerIcon from '../images/icons/icon-engineer.png'
import BusinessmanIcon from '../images/icons/icon-businessman.png'

const panelData = {
  genius: {
    icon: GeniusIcon,
    heading: 'Super-Genius Intelligence',
    body:
      'Quite apart from the powers granted him by the suit, Tony Stark is far more than a mechanical engineering prodigy who graduated from the Massachusetts Institute of Technology with honors at the age of 17.'
  },
  tactician: {
    icon: TacticianIcon,
    heading: 'Expert Tactician',
    body:
      'He is a brilliant tactician capable of quickly formulating battle strategies and new plans if the situation changes, like being able to elaborate complex plans in order to defeat different enemies.'
  },
  engineer: {
    icon: EngineerIcon,
    heading: 'Master Engineer',
    body:
      'He is an excellent engineer and mechanic capable of fixing almost any, if not all machinery.'
  },
  businessman: {
    icon: BusinessmanIcon,
    heading: 'Master Businessman',
    body:
      "Stark is extremely well-respected in the business world, able to command people's attentions when he speaks on economic matters.  He has built up several multi-million dollar companies from virtually nothing."
  }
}

const CopyPanel = () => {
  return (
    <PanelWrapper className="copy-panel">
      {Object.keys(panelData).map((key, i) => (
        <PanelContainer key={i}>
          <img src={panelData[key].icon} alt="" />
          <h4>{panelData[key].heading}</h4>
          <p>{panelData[key].body}</p>
        </PanelContainer>
      ))}
    </PanelWrapper>
  )
}

const PanelWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 0.2rem;

  @media (min-width: 768px) {
    flex: 0 1 50%;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
  }
`

const PanelContainer = styled.div`
  width: 100%;

  img {
    width: auto;
    height: 2.2rem;
    padding: 2rem 0 1rem;
  }

  h4 {
    line-height: 1.5rem;
    padding-bottom: 0.5rem;
    font-weight: 100;
  }

  p {
    color: #808080;
    font-size: 0.875rem;
  }

  @media (min-width: 768px) {
    width: calc(50% - 2rem);

    &:first-child {
      order: 0;
    }

    &:nth-child(2) {
      order: 3;
    }

    &:nth-child(odd) {
      margin-bottom: 2rem;
    }

    img {
      width: auto;
      padding: 0 0 1rem;
    }

    p {
      color: #808080;
      font-size: 1rem;
    }
  }

  @media (min-width: 1200px) {
    width: calc(50% - 4rem);

    img {
      height: 3.3rem;
    }

    h4 {
      font-size: 1.25rem;
    }
  }
`

export default CopyPanel
