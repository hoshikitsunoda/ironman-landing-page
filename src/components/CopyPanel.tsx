import React from 'react'
import { panelData } from '../data/seed'

import styled from 'styled-components'

const CopyPanel: React.FC = () => {
  return (
    <PanelWrapper className="copy-panel">
      {Object.keys(panelData).map((key: string, i: number) => (
        <PanelContainer key={i}>
          <img src={panelData[key].icon} alt={key} />
          <h4>{panelData[key].heading}</h4>
          <p>{panelData[key].body}</p>
        </PanelContainer>
      ))}
    </PanelWrapper>
  )
}

const PanelWrapper = styled('div')`
  display: flex;
  flex-direction: column;
  padding: 0.2rem;

  @media (min-width: 768px) {
    flex: 0 1 50%;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
  }
`

const PanelContainer = styled('div')`
  width: 100%;

  img {
    width: auto;
    height: 2.2rem;
    padding: 2rem 0 1rem;
  }

  h4 {
    line-height: 1.5rem;
    padding-bottom: 0.5rem;
    font-weight: 100;
  }

  p {
    color: #808080;
    font-size: 0.875rem;
  }

  @media (min-width: 768px) {
    width: calc(50% - 2rem);

    &:first-child {
      order: 0;
    }

    &:nth-child(2) {
      order: 3;
    }

    &:nth-child(odd) {
      margin-bottom: 2rem;
    }

    img {
      width: auto;
      padding: 0 0 1rem;
    }

    p {
      color: #808080;
      font-size: 1rem;
    }
  }

  @media (min-width: 1200px) {
    width: calc(50% - 4rem);

    img {
      height: 3.3rem;
    }

    h4 {
      font-size: 1.25rem;
    }
  }
`

export default CopyPanel
