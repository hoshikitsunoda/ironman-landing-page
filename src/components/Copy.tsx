import React from 'react'
import CopyPanel from './CopyPanel'

import styled from 'styled-components'

interface CharacterDataProps {
  characterData: { thumbnail: { path: string; extension: string } }
}

const Copy: React.FC<CharacterDataProps> = (props: CharacterDataProps) => {
  const imgSrcPath: string =
    props.characterData.thumbnail && props.characterData.thumbnail.path
  const imgSrcExtension: string =
    props.characterData.thumbnail && props.characterData.thumbnail.extension
  const imgSrc: string = `${imgSrcPath}.${imgSrcExtension}`
  return (
    <CopyWrapper className="copy-wrapper">
      <HeadingWrapper className="text-wrapper heading">
        <p>Contrary to popular belief, he knows exactly what heâ€™s doing.</p>
      </HeadingWrapper>
      <CopyContentWrapper className="copy-content-wrapper">
        <ImageWrapper className="image-wrapper">
          <img src={imgSrc} alt="thumbnail" className="thumbnail" />
        </ImageWrapper>
        <CopyPanel />
      </CopyContentWrapper>
    </CopyWrapper>
  )
}

const CopyWrapper = styled('div')`
  padding: 0 1.2rem;
  max-width: 1440px;
  margin: 0 auto;

  .thumbnail {
    width: 100%;
  }

  @media (min-width: 768px) {
    padding: 0 3rem;
  }

  @media (min-width: 1024px) {
    padding: 0 1.2rem;
  }
`

const HeadingWrapper = styled('div')`
  text-align: center;
  padding: 3rem 2.5rem;

  p {
    font-size: 1.5rem;
    line-height: 2.2rem;
  }

  @media (min-width: 768px) {
    p {
      font-size: 2rem;
    }
  }
`

const CopyContentWrapper = styled('div')`
  @media (min-width: 1024px) {
    display: flex;
    justify-content: center;
    align-items: flex-start;
  }
`

const ImageWrapper = styled('div')`
  @media (min-width: 768px) {
    padding: 0 0 2rem 0;
  }

  @media (min-width: 1024px) {
    flex: 0 1 35%;
    padding: 0 5rem 0 0;
  }
`

export default Copy
