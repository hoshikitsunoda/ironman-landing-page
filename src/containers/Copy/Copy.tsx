import React from 'react'
import CopyPanel from '../../components/CopyPanel/CopyPanel'
import { CopyDataProps } from '../../models/Props'

import styled from 'styled-components'
import * as Styled from '../../components/styled'

const Copy: React.FC<CopyDataProps> = (props: CopyDataProps) => {
  const { thumbnail } = props.characterData

  const imgSrcPath: string = thumbnail && thumbnail.path
  const imgSrcExtension: string = thumbnail && thumbnail.extension
  const imgSrc: string = `${imgSrcPath}.${imgSrcExtension}`

  return (
    <CopyWrapper className="copy-wrapper">
      <HeadingWrapper className="text-wrapper heading">
        <p>Contrary to popular belief, he knows exactly what heâ€™s doing.</p>
      </HeadingWrapper>
      <CopyContentWrapper className="copy-content-wrapper">
        <ImageWrapper className="image-wrapper">
          <img src={imgSrc} alt="thumbnail" className="thumbnail" />
        </ImageWrapper>
        <CopyPanel />
      </CopyContentWrapper>
    </CopyWrapper>
  )
}

const CopyWrapper = styled('div')`
  padding: 0 1.2rem;
  max-width: 1440px;
  margin: 0 auto;

  .thumbnail {
    width: 100%;
  }

  @media ${(props) => props.theme.device.tablet} {
    padding: 0 3rem;
  }

  @media ${(props) => props.theme.device.laptop} {
    padding: 0 1.2rem;
  }
`

const HeadingWrapper = styled('div')`
  text-align: center;
  padding: 3rem 2.5rem;

  p {
    font-size: 1.5rem;
    line-height: 2.2rem;
  }

  @media ${(props) => props.theme.device.tablet} {
    p {
      font-size: 2rem;
    }
  }
`

const CopyContentWrapper = styled('div')`
  @media ${(props) => props.theme.device.laptop} {
    ${Styled.flexbox('row', 'center', 'flex-start')}
  }
`

const ImageWrapper = styled('div')`
  @media ${(props) => props.theme.device.tablet} {
    padding: 0 0 2rem 0;
  }

  @media ${(props) => props.theme.device.laptop} {
    flex: 0 1 35%;
    padding: 0 5rem 0 0;
  }
`

export default Copy
