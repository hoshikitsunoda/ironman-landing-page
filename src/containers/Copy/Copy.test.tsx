import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import Copy from './Copy'
import CopyPanel from '../../components/CopyPanel/CopyPanel'
import copyPropsMockObj from '../../__test__/mocks/copyProps-mock.json'
import copyHeading from '../../__test__/mocks/copyHeading-mock.json'

import { ThemeProvider } from 'styled-components'
import * as Styled from '../../components/styled'

describe('<Copy />', () => {
  const { path, extension } = copyPropsMockObj.characterData.thumbnail
  let wrapper: ReactWrapper
  beforeEach(() => {
    wrapper = mount(
      <ThemeProvider theme={Styled.theme}>
        <Copy {...copyPropsMockObj} />
      </ThemeProvider>
    )
  })

  it('should render component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render heading paragraph', () => {
    expect(wrapper.find('.heading').hostNodes()).toHaveLength(1)
    expect(wrapper.find('.heading').find('p').text()).toEqual(
      copyHeading.heading
    )
  })

  it('should render thumbnail image', () => {
    expect(wrapper.find('.thumbnail')).toHaveLength(1)
    expect(wrapper.find('.thumbnail').props().src).toEqual(
      `${path}.${extension}`
    )
  })

  it('should render CopyPanel component', () => {
    expect(wrapper.find(CopyPanel)).toHaveLength(1)
  })
})
