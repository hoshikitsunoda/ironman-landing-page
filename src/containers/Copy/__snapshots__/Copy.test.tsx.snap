// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copy /> should render component 1`] = `ReactWrapper {}`;
